okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -918,7 +918,7 @@ public Builder newBuilder() {
    * @throws MalformedURLException if there was a non-host related URL issue
    * @throws UnknownHostException if the host was invalid
    */
-  static HttpUrl getChecked(String url) throws MalformedURLException, UnknownHostException {
+  public static HttpUrl getChecked(String url) throws MalformedURLException, UnknownHostException {
```

Comment:
I don't think we should just expose this with two checked exceptions. I would rather see a variant which throws IllegalArgumentException for all malformed input.

Reply:
IllegalArgumentException is RuntimeException.
Added check for null and empty string

Reply:
I know. The desire in the issie is for a method that throws any exception, not necessarily a checked one. And I think a checked one is inappropriate for this case.